File: job-portal_frontend/src/app/services/job.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';

const BASE_URL = 'http://localhost:8080/api/jobs';

@Injectable({ providedIn: 'root' })
export class JobService {
  constructor(private http: HttpClient) {}

  getAllJobs(): Observable<any> {
    return this.http.get(`${BASE_URL}`);
  }

  searchJobs(keyword: string): Observable<any> {
    return this.http.get(`${BASE_URL}/search`, { params: { keyword } });
  }

  filterJobs(filters: { category?: string, location?: string, experience?: string }): Observable<any> {
    let params = new HttpParams();
    if (filters.category) params = params.set('category', filters.category);
    if (filters.location) params = params.set('location', filters.location);
    if (filters.experience) params = params.set('experience', filters.experience);
    return this.http.get(`${BASE_URL}/filter`, { params });
  }

  getRecruiterJobs(): Observable<any> {
    return this.http.get(`${BASE_URL}/recruiter`);
  }

  getJobById(id: number): Observable<any> {
    return this.http.get(`${BASE_URL}/${id}`);
  }

  createJob(job: any): Observable<any> {
    return this.http.post(`${BASE_URL}`, job);
  }

  updateJob(id: number, job: any): Observable<any> {
    return this.http.put(`${BASE_URL}/${id}`, job);
  }

  deleteJob(id: number): Observable<any> {
    return this.http.delete(`${BASE_URL}/${id}`);
  }
    getJobsByRecruiter(): Observable<any> {
    return this.http.get(`${BASE_URL}/recruiter`);
  }
}

File: job-portal_backend/src/main/java/com/JobPortal/job_portal/services/JobService.java
package com.JobPortal.job_portal.services;

import com.JobPortal.job_portal.dtos.JobDTO;
import com.JobPortal.job_portal.models.Job;

import java.util.List;

public interface JobService {
    JobDTO createJob(JobDTO jobDTO);
    JobDTO getJobById(Long id);
    List<JobDTO> getAllJobs();
    List<JobDTO> getJobsByFilters(String category, String location, String experience);
    List<JobDTO> searchJobs(String keyword);
    List<JobDTO> getRecruiterJobs();
    JobDTO updateJob(Long id, JobDTO jobDTO);
    void deleteJob(Long id);
    Job getJobEntityById(Long id);
}